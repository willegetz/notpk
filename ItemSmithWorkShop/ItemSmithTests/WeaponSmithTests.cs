using System;
using System.Text;
using System.Collections.Generic;
using System.Linq;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using ApprovalTests.Reporters;
using ItemSmithWeaponSmith;
using ApprovalTests;
using ItemSmithWorkShop;

namespace ItemSmithTests
{
	[TestClass]
	[UseReporter(typeof(DiffReporter))]
	public class WeaponSmithTests
	{
		[TestMethod]
		public void TestDisplayDagger()
		{
			Dagger dagger = new Dagger();
			dagger.WeaponName("Dagger");
			dagger.WeaponCost(2);

			Approvals.Approve(dagger);
		}

		[TestMethod]
		public void TestProperNameDagger()
		{
			Dagger dagger = new Dagger();
			dagger.WeaponName("One-eyed Bart's Shiv");
			dagger.WeaponCost(2);

			Approvals.Approve(dagger);
		}

		[TestMethod]
		public void TestDmChangesPrice()
		{
			Dagger dagger = new Dagger();
			dagger.WeaponName("Expensive Dagger");
			dagger.WeaponCost(10);

			Approvals.Approve(dagger);
		}

		[TestMethod]
		public void TestWeaponIsMasterWork()
		{
			Dagger dagger = new Dagger();
			dagger.WeaponName("Dagger");
			dagger.WeaponCost(2);

			dagger.IsMasterworkQualifier(true);

			Approvals.Approve(dagger);
		}

		[TestMethod]
		public void TestSteelProperty()
		{
			SteelMaterial steel = new SteelMaterial();

			Approvals.Approve(steel);
		}

		[TestMethod]
		public void TestAdamantineProperty()
		{
			ItemSmithWeaponSmith.AdamantineMaterial adamantine = new ItemSmithWeaponSmith.AdamantineMaterial();

			Approvals.Approve(adamantine);
		}

		[TestMethod]
		public void TestDagger()
		{
			Dagger dagger = new Dagger();

			dagger.WeaponName("Dagger");
			dagger.WeaponCost(2);

			Approvals.Approve(dagger);
		}

		[TestMethod]
		public void TestSteelPropertyOnDagger()
		{
			MaterialComponent steel = new SteelMaterial();
			Dagger dagger = new Dagger(steel);

			dagger.WeaponName("Dagger");
			dagger.WeaponCost(2);

			Approvals.Approve(dagger);
		}

		[TestMethod]
		public void TestAdamantinePropertyOnDagger()
		{
			MaterialComponent adamantine = new AdamantineMaterial();
			Dagger dagger = new Dagger(adamantine);

			dagger.WeaponName("Dagger");
			dagger.WeaponCost(2);

			Approvals.Approve(dagger);
		}

		[TestMethod]
		public void TestMaterialPassedIntoConstructor()
		{
			MaterialComponent steel = new SteelMaterial();
			Dagger dagger = new Dagger(steel);
			dagger.WeaponName("Dagger");
			dagger.WeaponCost(2);

			Approvals.Approve(dagger);
		}

		[TestMethod]
		public void TestNullMaterialPassedIntoConstructor()
		{
			Dagger dagger = new Dagger(null);
			dagger.WeaponName("Dagger");
			dagger.WeaponCost(2);

			Approvals.Approve(dagger);
		}

		[TestMethod]
		public void TestAdamantineMaterialPassedIntoConstructor()
		{
			MaterialComponent adamantine = new AdamantineMaterial();
			Dagger dagger = new Dagger(adamantine);
			dagger.WeaponName("Dagger");
			dagger.WeaponCost(2);

			Approvals.Approve(dagger);
		}

		[TestMethod]
		public void TestSimpleDaggerObject()
		{
			DaggerMedium simpleDagger = new DaggerMedium();

			Approvals.Approve(simpleDagger);
		}

		[TestMethod]
		public void TestDaggerDroppedIntoSteel()
		{
			DaggerMedium simpleDagger = new DaggerMedium();
			new WeaponSteel(simpleDagger);

			Approvals.Approve(simpleDagger);
		}

		[TestMethod]
		public void TestDaggerDroppedIntoSteelProperlyNamed()
		{
			DaggerMedium simpleDagger = new DaggerMedium();
			simpleDagger.WeaponName = "Slashy Blade of Happiness";

			new WeaponSteel(simpleDagger);

			Approvals.Approve(simpleDagger);
		}

		[TestMethod]
		public void TestMasterworkDagger()
		{
			DaggerMedium simpleDagger = new DaggerMedium();
			simpleDagger.IsMasterworkQualifier(true);

			Approvals.Approve(simpleDagger);
		}

		[TestMethod]
		public void TestDaggerDroppedIntoAdamantine()
		{
			DaggerMedium dagger = new DaggerMedium();
			new WeaponAdamantine(dagger);

			Approvals.Approve(dagger);
		}

		[TestMethod]
		public void TestDaggerDroppedIntoDarkwood()
		{
			DaggerMedium dagger = new DaggerMedium();
			new WeaponDarkwood(dagger);

			Approvals.Approve(dagger);
		}

		[TestMethod]
		public void TestDaggerDroppedIntoColdIron()
		{
			DaggerMedium dagger = new DaggerMedium();
			new WeaponColdIron(dagger);

			Approvals.Approve(dagger);
		}

		[TestMethod]
		public void TestMasterworkDaggerDroppedIntoColdIron()
		{
			DaggerMedium dagger = new DaggerMedium();
			dagger.IsMasterworkQualifier(true);
			new WeaponColdIron(dagger);

			Approvals.Approve(dagger);
		}

		[TestMethod]
		public void TestDaggerDroppedIntoMithral()
		{
			DaggerMedium dagger = new DaggerMedium();
			dagger.IsMasterworkQualifier(true);
			new WeaponMithral(dagger);

			Approvals.Approve(dagger);
		}

		[TestMethod]
		public void TestMasterworkDaggerDroppedIntoMithral()
		{
			DaggerMedium dagger = new DaggerMedium();
			dagger.IsMasterworkQualifier(true);
			new WeaponMithral(dagger);

			Approvals.Approve(dagger);
		}

		[TestMethod]
		public void TestDaggerDroppedIntoAlchemicalSilver()
		{
			DaggerMedium dagger = new DaggerMedium();
			new WeaponAlchemicalSilver(dagger);

			Approvals.Approve(dagger);
		}

		[TestMethod]
		public void TestMasterworkDaggerDroppedIntoAlchemicalSilver()
		{
			DaggerMedium dagger = new DaggerMedium();
			dagger.IsMasterworkQualifier(true);
			new WeaponAlchemicalSilver(dagger);

			Approvals.Approve(dagger);
		}

		[TestMethod]
		public void TestSmallSizeDagger()
		{
			DaggerSmall dagger = new DaggerSmall();
			
			Approvals.Approve(dagger);
		}
	}
}
