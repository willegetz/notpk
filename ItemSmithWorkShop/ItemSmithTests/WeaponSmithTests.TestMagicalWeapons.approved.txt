+1 Dagger (Medium) [Masterwork]
Melee Weapon
Light Weapon Proficiency
To Hit: +1
Damage: 1d4 +1 [19-20 x2] Piercing or Slashing, Magic
Hardness: 11
Hit Points: 3
Weight: 1 pound(s)
2302 gold pieces
Creation Costs
	Caster Level: 3
	Days to Create: 2
	Experience Cost: 80 experience points
	Raw Material Cost: 1000 gold pieces

Weapon Text:
	The dagger is a common secondary weapon. You get a +2 bonus on
	Sleight of Hand checks made to conceal a dagger on your body.
	This dagger is masterwork quality!


+2 Dagger (Small) [Masterwork]
Melee Weapon
Light Weapon Proficiency
To Hit: +2
Damage: 1d3 +2 [19-20 x2] Piercing or Slashing, Magic
Hardness: 7
Hit Points: 3
Weight: 0.5 pound(s)
8302 gold pieces
Creation Costs
	Caster Level: 6
	Days to Create: 8
	Experience Cost: 320 experience points
	Raw Material Cost: 4000 gold pieces

Weapon Text:
	The dagger is a common secondary weapon. You get a +2 bonus on
	Sleight of Hand checks made to conceal a dagger on your body.
	This dagger is masterwork quality!


+3 Dagger (Large) [Masterwork]
Melee Weapon
Light Weapon Proficiency
To Hit: +3
Damage: 1d6 +3 [19-20 x2] Piercing or Slashing, Magic
Hardness: 23
Hit Points: 7
Weight: 2 pound(s)
18304 gold pieces
Creation Costs
	Caster Level: 9
	Days to Create: 18
	Experience Cost: 720 experience points
	Raw Material Cost: 9000 gold pieces

Weapon Text:
	The dagger is a common secondary weapon. You get a +2 bonus on
	Sleight of Hand checks made to conceal a dagger on your body.
	This dagger is masterwork quality!


+4 Dagger (Tiny) [Masterwork]
Melee Weapon
Light Weapon Proficiency
To Hit: +4
Damage: 1d2 +4 [19-20 x2] Piercing or Slashing, Magic
Hardness: 6.5
Hit Points: 5
Weight: 0.5 pound(s)
32302 gold pieces
Creation Costs
	Caster Level: 12
	Days to Create: 32
	Experience Cost: 1280 experience points
	Raw Material Cost: 16000 gold pieces

Weapon Text:
	The dagger is a common secondary weapon. You get a +2 bonus on
	Sleight of Hand checks made to conceal a dagger on your body.
	This dagger is masterwork quality!


+5 Dagger (Huge) [Masterwork]
Melee Weapon
Light Weapon Proficiency
To Hit: +5
Damage: 1d8 +5 [19-20 x2] Piercing or Slashing, Magic
Hardness: 45
Hit Points: 13
Weight: 2 pound(s)
50304 gold pieces
Creation Costs
	Caster Level: 15
	Days to Create: 50
	Experience Cost: 2000 experience points
	Raw Material Cost: 25000 gold pieces

Weapon Text:
	The dagger is a common secondary weapon. You get a +2 bonus on
	Sleight of Hand checks made to conceal a dagger on your body.
	This dagger is masterwork quality!


